generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/types/_FieldSet.kt

```diff
@@ -45,9 +48,19 @@ private object FieldSetCoercing : Coercing<FieldSet, String> {
     override fun parseValue(input: Any): FieldSet = parseLiteral(input)
 
     override fun parseLiteral(input: Any): FieldSet =
-        if (input !is StringValue) {
-            throw CoercingParseLiteralException("Cannot parse $input to FieldSet. Expected AST type 'StringValue' but was '${input.javaClass.simpleName}'.")
+        when (input) {
+            is FieldSet -> input
+            is StringValue -> FieldSet::class.constructors.first().call(input.value)
+            else -> {
+                throw CoercingParseLiteralException("Cannot parse $input to FieldSet. Expected AST type 'StringValue' but was '${input.javaClass.simpleName}'.")
+            }
+        }
+
+    override fun valueToLiteral(input: Any): Value<out Value<*>> {
+        if (input is FieldSet) {
```

Comment:
for consistency with ^ method, would it worth to use `when` ?

Reply:
for consistency changed to expression body syntax